template/CMakeLists.txt

```diff
@@ -1,8 +1,30 @@
-cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR)
-project(PROJECT_NAME)
+cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
+project(PROJECT_NAME VERSION 0.0.1 LANGUAGES CXX)
 
 include_directories(${PROJECT_SOURCE_DIR}/include)
 
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra -pedantic -Wold-style-cast -Wuninitialized -Wunreachable-code -Wstrict-overflow=3 -D_FORTIFY_SOURCE=2 -ffunction-sections -fdata-sections")
+
+  set(LINKER_FLAGS "-Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--gc-sections")
```

Comment:
As far as I can tell and the docs state it's boolean, either > 0 or <= 0. I worked through Ulrich Drepper's DSO paper a few years ago and derived those linker flags like the gnu style hashtable, O1 and so on.

References:
- https://www.akkadia.org/drepper/dsohowto.pdf (makes for a good weekend read)